frontends/web/src/routes/account/send/send.tsx

```diff
@@ -510,13 +510,31 @@ class Send extends Component<Props, State> {
       sendAll: false,
       fiatAmount: ''
     } as Pick<State, keyof State>;
-    if (amount) {
-      updateState['amount'] = amount;
+
+    const coinCode = this.getAccount()!.coinCode;
+    if (amount && coinCode === 'btc') {
+      // needed to convert to sats if sats mode is enabled.
+      parseExternalBtcAmount(amount).then(result => {
+        if (result.success) {
+          updateState['amount'] = result.amount;
+          this.setState(updateState, () => {
+            this.convertToFiat(this.state.amount);
+            this.validateAndDisplayFee(true);
+          });
```

Comment:
if you make parseQRResult an async function, the control flow here will become easier i think - you can await the result of `parseeExternalBtcAmount` and not duplicate these lines.

Reply:
Right! I was a bit afraid of inadvertently modifying previous behavior, but looks much nicer now :+1: